### DISCUSSION TITLE
future of augustus formula?

### DISCUSSION THREAD

**Discussion by mitchblank** [Homebrew/discussions#368] [Everyday usage]
I took a look at upgrading augustus from 3.3.3 to 3.4.0 to track upstream https://github.com/Gaius-Augustus/Augustus/releases   A lot has changed so I had to take some new dependencies like `mysql-client` and `mysql++`.  Eventually, though I hit a wall -- it now also requires `lp_solve` to build.

`lp_solve` isn't a formula that the main homebrew repo supports, but it is part of brewsci: https://github.com/brewsci/homebrew-science/blob/master/Formula/lp_solve.rb

So should we just send the augustus formula brewsci's way and remove it from the main homebrew tap?

Just to be clear, I was just looking into this formula as a general bit of homebrew maintenance; I don't know anything about the package or how it is used myself.  It obviously is science-oriented so brewsci could be a more loving home for it anyway.

ps: fwiw here are the partial changes I had made to the formula before hitting the lp_solve problem in case anyone else wants to pick it up:
```
diff --git a/Formula/augustus.rb b/Formula/augustus.rb
index d2bf8253fb..b078929ee9 100644
--- a/Formula/augustus.rb
+++ b/Formula/augustus.rb
@@ -1,10 +1,9 @@
 class Augustus < Formula
   desc "Predict genes in eukaryotic genomic sequences"
   homepage "https://bioinf.uni-greifswald.de/augustus/"
-  url "https://github.com/Gaius-Augustus/Augustus/releases/download/v3.3.3/augustus-3.3.3.tar.gz"
-  sha256 "4cc4d32074b18a8b7f853ebaa7c9bef80083b38277f8afb4d33c755be66b7140"
+  url "https://github.com/Gaius-Augustus/Augustus/releases/download/v3.4.0/augustus-3.4.0.tar.gz"
+  sha256 "246772737862d16755ab0197bb1d4656cafb31db9b38c2cd991a8e8db6f116c8"
   license "Artistic-1.0"
-  revision 1
   head "https://github.com/Gaius-Augustus/Augustus.git"
 
   livecheck do
@@ -23,6 +22,10 @@ class Augustus < Formula
   depends_on "boost" => :build
   depends_on "bamtools"
   depends_on "gcc"
+  depends_on "mysql++"
+  depends_on "mysql-client"
+
+  uses_from_macos "sqlite"
 
   def install
     # Avoid "fatal error: 'sam.h' file not found" by not building bam2wig
@@ -30,12 +33,15 @@ class Augustus < Formula
 
     # Fix error: api/BamReader.h: No such file or directory
     inreplace "auxprogs/bam2hints/Makefile",
-      "INCLUDES = /usr/include/bamtools",
-      "INCLUDES = #{Formula["bamtools"].include/"bamtools"}"
+      "INCLUDES = -I/usr/include/bamtools",
+      "INCLUDES = -I#{Formula["bamtools"].include/"bamtools"}"
     inreplace "auxprogs/filterBam/src/Makefile",
       "BAMTOOLS = /usr/include/bamtools",
       "BAMTOOLS= #{Formula["bamtools"].include/"bamtools"}"
 
+    inreplace "src/Makefile", "-I/usr/include/mysql++", "-I#{Formula["mysql++"].include}/mysql++"
+    inreplace "src/Makefile", "-I/usr/include/mysql ", "-I#{Formula["mysql-client"].include}/mysql "
+
     # Prevent symlinking into /usr/local/bin/
     inreplace "Makefile", %r{ln -sf.*/usr/local/bin/}, "#ln -sf"
 
```

**Comment by carlocab** [comment]
Two other options might be to:
1. Add an `lp_solve` formula
2. Vendor `lp_solve` in the `augustus` formula as a `resource`

**Comment by SMillerDev** [comment]
I think the best way is to migrate lp_solve into homebrew-core. Brewsci can just use the upstream version and we can migrate augustus. Win-win iyam.

**Comment by mitchblank** [comment]
The thing I don't know is what the process is (either at a technical or bureaucratic level) for migrating formula between homebrew and brewsci.  Is there some established coordination between the projects?

I don't have any huge interest in this formula, as I said.  I am just raising the issue found while researching the version update.

**Reply by Rylan12** [reply]
While there's no official connection, the folks who run brewsci are all Homebrew maintainers so you probably won't have much of a bureaucratic issue. I'd say just open a PR explaining and linking here and I bet it will be fine. And if not, the worst that can happen is that the PR is closed.

See https://docs.brew.sh/Migrating-A-Formula-To-A-Tap for the technical details.

**Reply by carlocab** [reply]
There's even a current example:

https://github.com/Homebrew/homebrew-core/pull/67822
https://github.com/brewsci/homebrew-science/pull/265
